README.md

```diff
@@ -464,6 +464,81 @@ repository, and `ZULIP_GIT_REF` to be any ref name in that repository
 
 Then stop and restart the container as described in the previous section.
 
+### Upgrading zulip/zulip-postgresql to 14
+
+These instructions assume that you have not changed the default
+Postgres data path (`/opt/docker/zulip/postgresql/data`) in your
+`docker-compose.yml`.  If you have changed it, please replace all
+occurences of `/opt/docker/zulip/postgresql/data` with your path.
+
+1. Make a backup of your Zulip Postgres data dir.
+
+2. Stop all Zulip containers, except the postgres one (e.g. use
+   `docker stop` and not `docker-compose stop`).
+
+3. Create a new (upgraded) Postgres container using a different data directory:
+
+```
+docker run -d \
+      --name postgresnew \
+      -e POSTGRES_DB=zulip \
+      -e POSTGRES_USER=zulip \
+      -e POSTGRES_PASSWORD=zulip \
+      -v /opt/docker/zulip/postgresql/new:/var/lib/postgresql/data:rw \
+      zulip/zulip-postgresql:14
+```
+
+4. Use `pg_dumpall` to dump all data from the existing Postgres container to the new Postgres container:
+
+```
+docker-compose exec database pg_dumpall -U zulip | \
+    docker exec -i postgresnew psql -U zulip
+```
+
+5. Stop and remove both Postgres containers:
+
+```
+docker-compose rm --stop database
+docker stop postgresnew
+docker rm postgresnew
+```
+
+6. Edit your `docker-compose.yml` to use the `zulip/zulip-postgresql:14` image for the `database` container.
+
+7. Replace the old Postgres data directory with upgraded data directory:
+
+```
+sudo mv /opt/docker/zulip/postgresql/data /opt/docker/zulip/postgresql/old
+sudo mv /opt/docker/zulip/postgresql/new /opt/docker/zulip/postgresql/data
+```
+
+8. Delete the old existing containers:
+
+```
+docker-compose rm
```

Comment:
Is there a way to add additional options to this to make it feel safer? 

Also, I assume this destroys any logs on the old postgres container? I doubt they're important to many folks, but I suppose we could add a brief remark about backing up any logs to step 1 if so.

Reply:
This was copy-paste from step 8 of the existing instructions. They can be deleted using docker rm deliberately, by instance ID, but that's not something we can enumerate consistently for instructions.